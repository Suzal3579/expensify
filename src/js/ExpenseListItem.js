import React from 'react';
import { NavLink } from 'react-router-dom';
import moment from 'moment';

const ExpenseListItem = ({ expenses }) => (
    <div>
        {expenses.map(({
            description, id, amount, note, createdAt,
        }) => (
            <div key={id}>
                <h3><NavLink to={`/editExpense/${id}`}> {description}</NavLink></h3>
                <p>{note}</p>
                <p>{amount}</p>
                <small>{moment(createdAt).format('MMMM Do, YYYY')}</small>
            </div>
        ))}
    </div>
);

export default ExpenseListItem;
