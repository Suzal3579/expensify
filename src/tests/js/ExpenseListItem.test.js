import React from 'react';
import toJSON from 'enzyme-to-json';
import { shallow } from 'enzyme';

import ExpenseListItem from '../../js/ExpenseListItem';

const expenses = [{
    id: 'one1',
    description: 'desc 1',
    note: 'Some note 1',
    amount: 200,
    createdAt: 0,
}, {
    id: 'two2',
    description: 'Some 2',
    note: 'Some note 2',
    amount: 1200,
    createdAt: 5,
}, {
    id: 'three3',
    description: 'Some desc 3',
    note: 'Some note 3',
    amount: 3200,
    createdAt: 55,
}];

test('', () => {
    const wrapper = shallow(<ExpenseListItem expenses={expenses} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
});
