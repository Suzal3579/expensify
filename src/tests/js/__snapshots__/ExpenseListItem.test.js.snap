// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div>
  <div
    key="one1"
  >
    <h3>
      <NavLink
        activeClassName="active"
        ariaCurrent="true"
        to="/editExpense/one1"
      >
         
        desc 1
      </NavLink>
    </h3>
    <p>
      Some note 1
    </p>
    <p>
      200
    </p>
    <small>
      January 1st, 1970
    </small>
  </div>
  <div
    key="two2"
  >
    <h3>
      <NavLink
        activeClassName="active"
        ariaCurrent="true"
        to="/editExpense/two2"
      >
         
        Some 2
      </NavLink>
    </h3>
    <p>
      Some note 2
    </p>
    <p>
      1200
    </p>
    <small>
      January 1st, 1970
    </small>
  </div>
  <div
    key="three3"
  >
    <h3>
      <NavLink
        activeClassName="active"
        ariaCurrent="true"
        to="/editExpense/three3"
      >
         
        Some desc 3
      </NavLink>
    </h3>
    <p>
      Some note 3
    </p>
    <p>
      3200
    </p>
    <small>
      January 1st, 1970
    </small>
  </div>
</div>
`;
